src/back/spirv/layout.rs

```diff
@@ -0,0 +1,216 @@
+use spirv::*;
+
+pub(crate) struct Module {
```

Comment:
Yes it is. As far as I understand from the spec, you have a Module which consists of two layouts: the Physical Layout and the Logical Layout. The Physical Layout is indeed also called Header, but I haven't found it called that in the spec instead of Physical Layout.

Reply:
Would be good to avoid collision with `crate::Module` semantics

Reply:
True, this `Module` struct is redundant and is only a wrapper around Physical- and LogicalLayout. I remove this and instead directly use the `PhysicalLayout`  and `LogicalLayout`.